### PR TITLE
DM: deinit initialized pci device when failed

### PR DESCRIPTION
**Pull request by zhijianli88**
if some pci devices behind gvt got failures when initlizing at init_pci(), gvt
instance have no chance to be destroied even acrn-dm exits.

NOTE: this patch can not work standalone, a following patch to kernel
side(643d40961cf: "vhm: init client->kthread_exit true") is requied as well,
otherwise it will stucks during destroying gvt instance.

Tracked-On: #1141
Signed-off-by: Li Zhijian <zhijianli88@163.com>

### CODE REVIEW

**Comment by lijinxia** [comment]
OK to verify

**Comment by dongyaozu** [comment]
@ywan170 @fyin1  can you review?

**Comment by anthonyzxu** [comment]
can we initiate gvt last?

File: devicemodel/hw/pci/core.c [hunk 1]
```diff
@@ -1270,6 +1272,29 @@ init_pci(struct vmctx *ctx)
...
+				ops = pci_emul_finddev(fi->fi_name);
+				assert(ops != NULL);
+				pci_emul_deinit(ctx, ops, bus, slot,
+				    func, fi);
+			}
+		}
+	}
+
+	return error;
 }
```
[the remark this answers is not in the document]
Comment (zhijianli88) [reply devicemodel/hw/pci/core.c]: Looks no need to do anything for create_gsi_sharing_groups

**Comment by jianlaipinan** [comment]
Ready to merge

**Pull request merged**